From 21c0cc198c994be76d38b6378c93340e1d7bbc59 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Thu, 3 Sep 2020 23:00:04 -0300
Subject: [PATCH] Remove static library support

Q_COREAPP_STARTUP_FUNCTION, which is used by ECM's generated translation
loader, gets automatically stripped when using a static library.

See
https://doc.qt.io/qt-5/qcoreapplication.html#Q_COREAPP_STARTUP_FUNCTION
.
---
 src/SeExpr2/UI/CMakeLists.txt | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/src/SeExpr2/UI/CMakeLists.txt b/src/SeExpr2/UI/CMakeLists.txt
index 3274a41..919b79e 100644
--- a/src/SeExpr2/UI/CMakeLists.txt
+++ b/src/SeExpr2/UI/CMakeLists.txt
@@ -160,20 +160,9 @@ if (Qt5_FOUND OR QT4_FOUND)
             ${SeExpr2Editor_QM_LOADER})
     endif()
 
-    if (WIN32)
-        add_library(SeExpr2Editor ${EDITOR_CPPS} ${EDITOR_MOC_SRCS}
-                    ${ANIMLIB_SRCS}
-                    ${editor_parser_cpp})
-        generate_export_header(SeExpr2Editor
-                               BASE_NAME SeExpr2Editor
-                               EXPORT_MACRO_NAME SeExpr2Editor
-                               EXPORT_FILE_NAME SeExpr2Editor_Export.h
-                               STATIC_DEFINE SeExpr2Editor_BUILT_AS_STATIC )
-    else()
-        add_library(SeExpr2Editor SHARED ${EDITOR_CPPS} ${EDITOR_MOC_SRCS}
-                    ${ANIMLIB_SRCS}
-                    ${editor_parser_cpp})
-    endif()
+    add_library(SeExpr2Editor SHARED ${EDITOR_CPPS} ${EDITOR_MOC_SRCS}
+                ${ANIMLIB_SRCS}
+                ${editor_parser_cpp})
 
     set_property(TARGET SeExpr2Editor PROPERTY VERSION ${SeExpr2_VERSION})
     set_property(TARGET SeExpr2Editor PROPERTY SOVERSION 3)
-- 
2.27.0.windows.1

